modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/getItemField.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {
+	FDS_ARRAY_FIELD_NAME_PARENT_SUFFIX,
+	FDS_NESTED_FIELD_NAME_DELIMITER,
+	FDS_NESTED_FIELD_NAME_PARENT_SUFFIX,
+} from '../constants';
+
+/**
+ * Drills down an object using the path
+ * @param path : string
+ * @param item : any
+ * @returns The part of the item indicated by the path
+ *
+ * path: 'a.b.c'
+ * item: {a: {b: {c: [1, 2, 3]}}}
+ * returns: [1, 2, 3]
+ *
+ * path: 'a.b*'
+ * item: {a: {b: {c: [1, 2, 3]}}}
+ * returns: {c: [1, 2, 3]}
+ *
+ * path: 'a[].b.c'
+ * item: {a: [{b: {c: 1, d: 3}}, {b: {c: 2, d: 3}}, {b: {c: 3, d: 3}}]}
+ * returns: [{c:1}, {c: 2}, {c: 3}]
```

Comment:
Shouldn't return here be `[1, 2, 3]`? The expectation is type of array items that matches type of the leaf, `c`, and that is a number.

Btw, love the examples docs! :heart:

Reply:
Returning `[1, 2, 3]` will be part of [`getLocalizedValue`](https://github.com/liferay-frontend/liferay-portal/pull/4135/files#diff-8fb8b733d3441cdfc6070de3afc7eab272f73cd50ff30d81388045401c9d571cR147) . Consider that the value can contain strings to be translated. 

Related test -> https://github.com/liferay-frontend/liferay-portal/pull/4135/files#diff-49ba7db4dc49e6466b818786bdabaff962cb9d0df69c1c281e73aa99d5180c87R233

Reply:
OK, makes sense :+1: